api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1694,18 +1694,72 @@ type AWSServiceEndpoint struct {
 }
 
 type AzurePlatformSpec struct {
+	// Credentials is the object containing the Azure credentials needed when creating and managing Azure cloud
+	// infrastructure resources
+	//
+	// +kubebuilder:validation:Required
 	Credentials corev1.LocalObjectReference `json:"credentials"`
-	// The cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
-	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
+
+	// Cloud is the cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+	//
+	// +kubebuilder:validation:Enum="";AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud;AzureStackCloud
```

Comment:
why allow empty "" value?

Reply:
I got that as a valid value from the OpenShift API vendor files - https://github.com/openshift/hypershift/blob/ac66d8f59abacf5e2650e0bc602c1cc8fc1bde79/vendor/github.com/openshift/api/config/v1/types_infrastructure.go#L581

Reply:
I see, I am wondering how does this improve the UX, when would a user set this to "" versus just leave the field undefined.
Both cases will default to PublicCloud, but the if explicitly set to "", the default value will not be directly visible in the spec.

Reply:
This is true. They were not defaulting it either. I will remove the `""`.